### ISSUE TITLE
Define models programmatically 

### ISSUE THREAD

**Issue by Utkarsh85** [vdemedes/mongorito#70]
I have a scenario where model names are defined by user and we get all models as an array strings. E.g.
["user","post","comment"] ......

How can I generate models using this array. E.g.

```javascript
var Mongorito = require('mongorito');
var Model = Mongorito.Model;

Mongorito.connect('localhost/db');

var arr=["user","post","comment"] ;
//Hypothetical
arr.map(function(val){class val extends Model {};})
```

**Comment by Utkarsh85** [comment]
**Got it**
```javascript
var arr=["user","post","comment"] ;
var models={};
arr.map(function(val){models[val]=Model.extend({
    collection: val,
});
});
```

**Comment by avivcarmis** [comment]
How do you plan on using it later?

There are some ways of doing that but it would probably not help you. For example you can do

```
arr.map(function(val){
  eval("class " + val + " extends Model {}");
});
```
As I said this would probably not help you since you can't do ``` new arr[0]()```, and on the other hand you definitely don't want to continue working with ```eval``` as it's really just an awful practice.

Another way to achieve that is programmatically inherit Model prototype, that might be helpful if you would like to create single instances of this classes each time.

Please explain what do you plan to do with those classes.

**Comment by Utkarsh85** [comment]
I am using Koa Framework and plan to use it like this:-
a) I have all the models in a '/model' folder. So '/model/User.js' will have all the attributes of the User model defined. There are others too. Eg. '/model/Post.js' , '/model/Comment.js', etc.
```javascript
// /mode/User.js
module.exports = {

  attributes: {
     name:'string',
     .......
  }
};
```
b) Now in './Model.js' I want to unify all the models into one nice api, which you can require in your '/Controllers/UserController.js' and other controllers without having to deal with Mongorito.
e.g. You can just do this
```javascript
// In '/Controllers/UserController.js'
var Models=require('../Model.js');
var user=new Model.User();
user.save();

//and
var post= new Model.Post();
post.save();
```

c) The './Model.js' currently looks like this
```javascript
"use strict";

var Mongorito = require('mongorito');
var MongoritoModel = Mongorito.Model;
var requireDir = require('require-dir');

Mongorito.connect('localhost/blog');


var userdefinedmodels = requireDir('../api/models');//Find all the user defined models
//Looks like this
//{User:{},Post:{},Comment:{}}

var Model={}; // The models object

for(var key in userdefinedmodels )
{
	Model[key]=class extends MongoritoModel{

	    collection() {
	       return key;
	    }

	};
}

module.exports=Model;
```
d) ***Problem***
Due to the use of loop in the above code, all Models' collection are being set to the last key in the loop.
[Refer](http://stackoverflow.com/questions/1451009/javascript-infamous-loop-issue)

**Comment by vdemedes** [comment]
I just added `Model.extend()` and published a new version `1.4.4` with this change, so this should solve your problem. You can now:

```js
var DynamicModel = Model.extend({
  collection: 'dynamic collection name'
});
```
